services/file_update.proto

```diff
@@ -31,37 +38,74 @@ import "timestamp.proto";
+ * Update the metadata, and/or replace the content, of a file in the
+ * Hedera File Service (HFS).
+ *
+ * Any field, other than `fileID` which is not set (i.e. is null)
+ * in this message SHALL be ignored.<br/>
+ * If the `keys` list for the identified file is an empty `KeyList`, then
+ * this message MUST NOT set any field except `expirationTime`.
+ *
+ * #### Signature Requirements
+ * Every `Key` in the `keys` list for the identified file MUST sign this
+ * transaction, if any field other than `expirationTime` is to be updated.<br/>
+ * See the [File Service](#FileService) specification for a detailed
+ * explanation of the signature requirements for all file transactions.
+ *
+ * ### Block Stream Effects
+ * None
  */
 message FileUpdateTransactionBody {
     /**
-     * The ID of the file to update
+     * A file identifier for the file to update.
+     * <p>
+     * This field is REQUIRED.
      */
     FileID fileID = 1;
 
     /**
-     * The new expiry time (ignored if not later than the current expiry)
+     * An expiration timestamp.
+     * <p>
+     * If set, this value MUST be strictly later than the existing
+     * `expirationTime` value.<br/>
+     * If set, this value SHALL replace the existing `expirationTime`.<br/>
+     * If this field is the only field set, then this transaction SHALL NOT
+     * require any signature other than the `payer` for the transaction.<br/>
+     * When the network consensus time exceeds the then-current
+     * `expirationTime`, the network SHALL expire the file.
      */
     Timestamp expirationTime = 2;
 
     /**
-     * The new list of keys that can modify or delete the file
+     * The new list of keys that "own" this file.
+     * <p>
+     * If set, every key in this `KeyList` MUST sign this transaction.<br/>
+     * If set, every key in the _previous_ `KeyList` MUST _also_
+     * sign this transaction.<br/>
+     * If this value is an empty `KeyList`, then the file SHALL be immutable
+     * after completion of this transaction.
      */
     KeyList keys = 3;
 
     /**
-     * The new contents that should overwrite the file's current contents
+     * An array of bytes.
+     * <p>
+     * This value, if set, SHALL _replace_ the existing file content.
+     * If this value is set to an empty byte array, the content of the file
+     * SHALL be unchanged.
+     * <blockquote>REVIEW NOTE<blockquote>
+     * This is a CHANGE from current behavior, but is REQUIRED because the
+     * "default" value for a PBJ `Bytes` is no longer `null`, but
+     * is `Bytes.EMPTY`. We may have to make a change here to enable clearing
+     * the content of a file post-modularization...
+     * </blockquote></blockquote>
      */
     bytes contents = 4;
 
     /**
-     * If set, the new memo to be associated with the file (UTF-8 encoding max 100 bytes)
+     * A short description of this file.
+     * <p>
+     * This value, if set, MUST NOT exceed 100 bytes when encoded as UTF-8.
```

Comment:
Should we call out configuration properties when applicable? This value must no exceed transaction.maxMemoUtf8Bytes which has a default value of 100 bytes.

Reply:
Updated.